test/core/event_engine/posix/posix_event_engine_connect_test.cc

```diff
@@ -0,0 +1,240 @@
+    return absl::UnknownError(
+        absl::StrCat("Error setsockopt(SO_REUSEPORT): ", std::strerror(errno)));
+  }
+
+  // Forcefully bind the new socket.
+  if (bind(server_socket, resolved_addr.address(), resolved_addr.size()) < 0) {
+    return absl::UnknownError(
+        absl::StrCat("Error bind: ", std::strerror(errno)));
+  }
+  // Set the new socket to listen for one active connection at a time.
+  // accept() is intentionally not called on the socket. This allows the
+  // connection queue to build up.
+  if (listen(server_socket, 1) < 0) {
+    return absl::UnknownError(
+        absl::StrCat("Error listen: ", std::strerror(errno)));
+  }
+  ret_sockets.push_back(server_socket);
+  // Create and connect client sockets until the connection attempt times out.
+  // Even if the backlog specified to listen is 1, the kernel continues to
+  // accept a certain number of SYN packets before dropping them. This loop
+  // attempts to identify the number of new connection attempts that will
+  // be allowed by the kernel before any subsequent connection attempts
+  // become pending indefinitely.
+  while (true) {
+    client_socket = socket(AF_INET6, SOCK_STREAM, 0);
+    setsockopt(client_socket, SOL_SOCKET, SO_REUSEADDR, &one, sizeof(one));
+    // Make fd non-blocking.
+    flags = fcntl(client_socket, F_GETFL, 0);
+    EXPECT_EQ(fcntl(client_socket, F_SETFL, flags | O_NONBLOCK), 0);
+
+    if (connect(client_socket,
+                const_cast<struct sockaddr*>(resolved_addr.address()),
+                resolved_addr.size()) == -1) {
+      if (errno == EINPROGRESS) {
+        struct pollfd pfd;
+        pfd.fd = client_socket;
+        pfd.events = POLLOUT;
+        pfd.revents = 0;
+        int ret = poll(&pfd, 1, 1000);
+        if (ret == -1) {
+          gpr_log(GPR_ERROR, "poll() failed during connect; errno=%d", errno);
+          abort();
+        } else if (ret == 0) {
+          // current connection attempt timed out. It indicates that the
+          // kernel will cause any subsequent connection attempts to
+          // become pending indefinitely.
+          ret_sockets.push_back(client_socket);
+          return ret_sockets;
+        }
+      } else {
+        gpr_log(GPR_ERROR, "Failed to connect to the server (errno=%d)", errno);
+        abort();
+      }
+    }
+    ret_sockets.push_back(client_socket);
+  }
+  return ret_sockets;
+}
+
+}  // namespace
+
+TEST(PosixEventEngineTest, IndefiniteConnectTimeoutOrRstTest) {
+  std::string target_addr = absl::StrCat(
+      "ipv6:[::1]:", std::to_string(grpc_pick_unused_port_or_die()));
+  auto resolved_addr = URIToResolvedAddress(target_addr);
+  std::shared_ptr<EventEngine> posix_ee = std::make_shared<PosixEventEngine>();
+  std::string resolved_addr_str =
+      SockaddrToString(&resolved_addr, true).value();
+  auto sockets = CreateConnectedSockets(resolved_addr);
+  grpc_core::Notification signal;
+  ASSERT_TRUE(sockets.ok());
```

Comment:
The returned status is never used, never shown. I believe any errors will show up as `Expected: true. Actual: false`. Consider instead doing the logging and asserts in the CreateConnectedSockets method itself.

Reply:
Fixed.